app/src/test/java/org/simple/clinic/registration/register/RegistrationLoadingScreenControllerTest.kt

```diff
@@ -21,61 +19,84 @@ import org.simple.clinic.user.registeruser.RegistrationResult.UnexpectedError
 import org.simple.clinic.widgets.ScreenCreated
 import org.simple.clinic.widgets.UiEvent
 
-@RunWith(JUnitParamsRunner::class)
 class RegistrationLoadingScreenControllerTest {
 
   private val userSession = mock<UserSession>()
   private val screen = mock<RegistrationLoadingScreen>()
-  lateinit var controller: RegistrationLoadingScreenController
-
   private val uiEvents = PublishSubject.create<UiEvent>()
 
+  private val controller = RegistrationLoadingScreenController(userSession)
+
   @Before
   fun setUp() {
-    controller = RegistrationLoadingScreenController(userSession)
-
     uiEvents
         .compose(controller)
         .subscribe { uiChange -> uiChange(screen) }
   }
 
-  @Parameters(method = "params for testing user registration")
   @Test
-  fun `when screen is created then user registration api should be called`(results: RegistrationResult) {
-    whenever(userSession.register()).doReturn(Single.just(results))
-    whenever(userSession.clearOngoingRegistrationEntry()).doReturn(Completable.complete())
+  fun `when retry button is clicked, then loader should show and register api should be called`() {
```

Comment:
The test implies that the loader should show. However, the verifications do not assert a loader being displayed. Am I missing something here?

Reply:
I didn't write this test right now, but I think the code got rearranged to show this as a change. I'll take a look at this test and see what's up.

Reply:
The `showLoader` function is being triggered on the screen via a side-effect when the retry button is clicked, but the function is private. I am guessing this was initially a part of the controller but was changed later without updating the test.

Will reword the test description to match.